src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -261,28 +262,27 @@ internal static void Remove(long bid)
     /// </summary>
     internal static class JvmBroadcastRegistry
     {
-        [ThreadStatic]
-        private static readonly List<JvmObjectReference> s_jvmBroadcastVariables =
-            new List<JvmObjectReference>();
+        private static ThreadLocal<List<JvmObjectReference>> s_jvmBroadcastVariables = 
```

Comment:
Are we still okay with broadcast variables declared in one thread, only being available in that thread, or do we want them shared across all threads ?

Reply:
I believe it should be thread local since that is how [pyspark](https://github.com/apache/spark/blob/master/python/pyspark/broadcast.py#L183) declares it, but please let me know if we think otherwise.

Reply:
Will we run into any issues when multiple broadcast variables are declared in separate threads and a udf needs access to them ?

Reply:
This is just on the driver side, to send the broadcast variables defined in a thread to the workers. I believe on the workers they will all be distributed so any udf that needs to access any broadcast should be able to. @imback82 please correct me if I'm wrong.

Reply:
This could be problematic in an environment where we don't control thread model, e.g., notebook integration. Since this is a blocking issue for the next release and there is still an on-going issue with notebook integration, we can document this as a known issue and fix it for the v0.13. Are you OK with this @suhsteve?

Reply:
Sure this can wait until next release.